aexpect/remote_door.py

```diff
@@ -207,10 +211,26 @@ def run_subcontrol(session, control_path, timeout=600, detach=False):
     :param bool detach: whether to detach from session (e.g. if running a
                         daemon or similar long-term utility)
     """
-    remote_control_path = os.path.join(REMOTE_CONTROL_DIR, os.path.basename(control_path))
-    remote.scp_to_remote(session.host, session.port, session.username, session.password,
+    # Linux VMs
+    if session.client == "ssh":
+        transfer_client = "scp"
+        transfer_port = 22
+        remote_control_path = os.path.join(REMOTE_CONTROL_DIR, os.path.basename(control_path))
+        python_binary = REMOTE_PYTHON_BINARY
+    # Windows VMs
+    elif session.client == "nc":
+        transfer_client = "rss"
+        transfer_port = 10023
+        # ..todo:: use REMOTE_CONTROL_DIR?
+        remote_control_path = "%TEMP%\\" + os.path.basename(control_path)
+        python_binary = session.cmd("where python", timeout=timeout,
+                                    print_func=logging.info).strip()
+    else:
+        raise NotImplementedError("run_subcontrol not implemented for client %s" % session.client)
+    remote.copy_files_to(session.host, transfer_client,
+                         session.username, session.password, transfer_port,
```

Comment:
Yes, this should result in the same no linux and something that looks windowish on windows. Is the session.client always set for the usual sessions? (I'm only faking it when testing) If not than it'd perhaps be nice to detect it rather than to expect it exists.

Reply:
The session object is of type `RemoteSession` which [always sets the client attribute](https://github.com/avocado-framework/aexpect/blob/76c6019a3d8afa3b2cc4de3bcc269e076e1724ac/aexpect/client.py#L1405). I am still concerned a bit about the value hard-coding here and the unfinished `REMOTE_CONTROL_DIR` implementation but this PR has been waiting for quite long in my shelves and it still adds windows capabilities so I guess we can improve on this front gradually depending on demand for windows hosts.

Reply:
I extended the windows functionality a bit leaving clearer todos regarding some further integration.